### PR TITLE
Support for more tooltip positions

### PR DESCRIPTION
**Pull request by vjsrinath**
Added support for new tooltip positions like "top left", "top right", "bottom right", "bottom left".

### CODE REVIEW

File: doc/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
   },
   "jspm": {
     "dependencies": {
-      "aurelia-bootstrap": "github:tochoromero/aurelia-bootstrap@master",
+      "aurelia-bootstrap": "github:appsflare/aurelia-bootstrap@master",
```
Comment (tochoromero) [inline doc/package.json]: This needs to point back to the real repo

File: doc/src/tooltip/tooltip.html [hunk 2]
```diff
@@ -47,8 +50,8 @@
               Dynamic Tooltip
             </button>
 
-            <p>I'm a paragraph with a <strong aubs-tooltip="text:Custom Class; class: tooltip, custom-tooltip">Tooltip</strong> with custom
-              class</p>
+            <p>I'm a paragraph with a <strong aubs-tooltip="text:Custom Class; class: tooltip, custom-tooltip">Tooltip</strong>              with custom class
```
Comment (tochoromero) [inline doc/src/tooltip/tooltip.html]: I like to keep lines shorter, I guess this was your formatter messing with the line breaks :p

File: build/tasks/build.js [hunk 3]
```diff
@@ -13,6 +13,14 @@ gulp.task('build-html', function () {
     .pipe(gulp.dest(paths.output + 'system'));
 });
 
+gulp.task('build-css', function () {
```
Comment (tochoromero) [inline build/tasks/build.js]: Why the build changes?

File: dist/amd/utils/tooltip-service.js [hunk 4]
```diff
@@ -20,28 +20,64 @@ define(['exports', 'tether'], function (exports, _tether) {
         }
     }
 
+    var attachmentModes = {
```
Comment (tochoromero) [inline dist/amd/utils/tooltip-service.js:23]: Nice, I really like this, it is much cleaner :)

File: build/tasks/build.js (hunk 3 above)
Comment (vjsrinath) [reply build/tasks/build.js]: As of **bootstrap v3.3.1** **.top-left, .top-right, .bottom-left** and **.bottom-right** classes are deprecated.
So, I thought we might have to ship minor css patches when the above mentioned classes are removed from bootstrap.

File: dist/amd/utils/tooltip-service.js (hunk 4 above)
Comment (vjsrinath) [reply dist/amd/utils/tooltip-service.js:23]: Thanks :)

**Comment by vjsrinath** [comment]
@tochoromero I have made the changes in package.json to point to actual repo and corrected the line break in tooltip.html.
Please check and let me know if there are any other feedback.

File: build/tasks/build.js (hunk 3 above)
Comment (tochoromero) [reply build/tasks/build.js]: I see, the problem is the library already supports Bootstrap 4, if we want to add this we need to also provide the implementation for Bootstrap 4...

**Comment by tochoromero** [comment]
My only problem with the Pull Request is that it only works for Bootstrap 3. I'm very hesitant to add features that only work for one version of Bootstrap.
In order to be able to merge this, we would need to provide the CSS to make this work on Bootstrap 4. Unfortunately, CSS is not my strong suit, if you would like to invest the time to create the CSS for Bootstrap 4 I would be extremely thankful :)

File: build/tasks/build.js (hunk 3 above)
Comment (vjsrinath) [reply build/tasks/build.js]: Is it okay if we ship custom css patches for bootstrap 4?

**Comment by vjsrinath** [comment]
Sure, In that case, I will add custom css patches to add support for bootstrap 4 and update the pull request.

**Comment by tochoromero** [comment]
Is there a way to do the css so they don't have to manually include it, so the library automatically includes it?
And we need to make sure the custom css doesn't mess up in Bootstrap 3 mode.
Thanks a lot

**Comment by vjsrinath** [comment]
Yes, we can make the module to include the css automatically. But there are some complications as well. Please take a look at the below issue.
[CSS imports](https://github.com/aurelia/bundler/issues/50)

I think it is better to leave the choice to the users whether they want it or not.

**Comment by tochoromero** [comment]
Unfortunately Bootstrap 4 Beta switched from using Tether to use Popper. The change is not a simple one, so I will have to rework this whole thing. However, I will borrow some of your improvements. Thank you very much for the contribution.

**Pull request closed without merging**